File: go/config/FastBoot.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// FastBoot 对应 config/FastBoot.py。
type FastBoot struct {
	*session.ApiGroup
}

// NewFastBoot 创建 FastBoot API 分组。
func NewFastBoot(s *session.Session) *FastBoot {
	return &FastBoot{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 fastboot/config.xml。
func (f *FastBoot) Config() (map[string]interface{}, error) {
	return f.S.Get("fastboot/config.xml", nil, "config")
}

File: src/config/fast_boot.rs
//! FastBoot config group (`config/FastBoot.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// FastBoot config group.
pub struct FastBootConfig<'a> {
    conn: &'a Connection,
}

impl<'a> FastBootConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        FastBootConfig { conn }
    }

    /// `fastboot/config.xml`. Fast boot configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/fastboot/config.xml")
    }
}
